ts/ui/menu/RadioCompare.ts

```diff
@@ -0,0 +1,77 @@
+/*************************************************************
+ *
+ *  Copyright (c) 2022 The MathJax Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/**
+ * @fileoverview  Implements a radio button with customizable comparator.
+ *
+ * @author v.sorge@mathjax.org (Volker Sorge)
+ */
+
+import {Menu} from 'mj-context-menu/js/menu.js';
+import {Radio} from 'mj-context-menu/js/item_radio.js';
+import {ParserFactory} from 'mj-context-menu/js/parser_factory.js';
+
+// Extend the radio buttons with a customizable comparator to work for CS
+// preferences
+export class RadioCompare extends Radio {
+
+
+  /**
+   * @override
+   */
+  protected role = 'menuitemradiocompare';
+
+  /**
+   * @override
+   */
+  public static fromJson(
+    _factory: ParserFactory,
+    {content: content, variable: variable, id: id, comparator: comparator}:
+    {content: string, variable: string, id: string,
+     comparator: (variable: string, id: string) => boolean}, menu: Menu): Radio {
```

Comment:
These lines are hard to parse (what are the arguments and what the types.  Perhaps moving `, menu: Menu` to a separate line and `): Radio {` to the line after that would help.

Reply:
I used `prettier` to make it more reader friendly. Have a look to see if this would be helpful to use in the future.

Reply:
Yes, that looks better.